src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -51,20 +87,28 @@ object ResourceMatcher {
       badConstraints.isEmpty
     }
 
-    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges match {
-      case None =>
-        log.warn("App ports are not available in the offer.")
-        None
+    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges
+
+    val basicResourceRoles = Map(
+      "cpu" -> cpuRoleOpt,
+      "disk" -> diskRoleOpt,
+      "mem" -> memRoleOpt
+    )
 
-      case x @ Some(portRanges) =>
-        log.debug("Met all constraints.")
-        x
+    if (basicResourceRoles.values.exists(_.exists(!_.matches))) {
+      val basicResourceString = basicResourceRoles.map {
+        case (resource, Some(scalarMatch)) =>
+          s"$resource $scalarMatch"
+        case (resource, None) =>
+          s"$resource not in offer"
+      }.mkString(", ")
+      log.info(s"Offer ${offer.getId.getValue}. Not all basic resources satisfied: $basicResourceString")
     }
 
     for {
-      cpuRole <- cpuRoleOpt
-      memRole <- memRoleOpt
-      diskRole <- diskRoleOpt
+      cpuRole <- cpuRoleOpt.flatMap(_.matchingRole)
+      memRole <- memRoleOpt.flatMap(_.matchingRole)
+      diskRole <- diskRoleOpt.flatMap(_.matchingRole)
```

Comment:
OK, we check for matching role here (and that the option is defined) but where do we check the value of `ScalarMatch.matches`?  It looks like that method is only invoked for the purposes of logging above.